### DISCUSSION TITLE
Accessing the JavaScript component instance: "Experimental support for decorators..." Error with VS Code

### DISCUSSION THREAD

**Discussion by trabdlkarim** [livewire/livewire#3629] [1. Help] [answered]
![js-experimental-error](https://user-images.githubusercontent.com/17558454/132146628-d0450ecc-759b-4492-8348-9f223eb604b8.png)
Hi there,
I'm trying to write a basic inline Javascript to access the component instance with the special **@this** blade directive using VS Code but getting the following error:
 
`Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option in your 'tsconfig' or 'jsconfig' to remove this warning.`

This seems to be a pretty popular problem and I have tried most of the proposed solutions  without any  success. So far I have set **experimentalDecorators** option to true in VC Code settings, and also added a tsconfig.json to my project root directory as proposed on several sites, but still getting the same error message.

Does anyone here have any idea on how to get rid of this error? I would really appreciate some comments and ideas that help me solve that.

**Comment by joshhanley** [comment, marked as answer]
@trabdlkarim I use vscode as my IDE and have never come across that.

Can you explain more about your setup and what it triggering that error?

Also you say it's a pretty popular problem, have you got any links to other forum threads?

Thanks!

**Reply by trabdlkarim** [reply]
@joshhanley I've just attached a screenshot of the error to the question. It seems to be triggered by the usage of the **@this** blade directive at line 681 inside the livewire component view.

 I just wanted to access my component instance in order to set some property from the frontend. As I said it's a common problem but in most cases the error occurs in a typical Javascript or Typescript  project. I've got some links to other forum threads, particularly [this one](https://stackoverflow.com/questions/38271273/experimental-decorators-warning-in-typescript-compilation) to Stackoverflow. 

And yes I have already tried the proposed solutions (in some answers they say to add a tsconfig.json or jsconfig.json with the **experimentalDecorators** option set to true), but it did not work for me. I don't know why. May be it is because of the fact that my project is not a js project.

**Reply by joshhanley** [reply]
@trabdlkarim thanks! I just tested that on my VSCode and it works happily and doesn't throw any errors.

Do you have any Laravel/Livewire extensions installed?

Currently I'm running:
- Laravel Blade formatter
- Laravel Blade Spacer
- Laravel Extra Intellisense
- Laravel goto view
- laravel-goto-controller
- Livewire Language Support

Maybe one of them is getting rid of the error for me.

Hope this helps!

**Reply by trabdlkarim** [reply]
@joshhanley actually I should have mentioned it, sorry for that. The error doesn't prevent my application from running, I mean Livewire and Laravel work perfectly. The problem is on VS Code's side. I couldn't find the right settings to tell VS Code to stop underlining the related line as an error. And to answer your question, NO I don't have any Laravel/Livewire extensions installed. I will try to install those extensions you listed and see whether that helps or not. Thanks for the effort ☺!

**Reply by trabdlkarim** [reply]
@joshhanley as you suggested installing Laravel Extra Intellisense and Livewire Language Support extensions solved the issue. I would like to thank you for your support. It has been really helpful, thanks a lot 😃😃😃!!!

**Reply by joshhanley** [reply]
@trabdlkarim awesome! No worries 😁
